File: orm/jkorm-mybatis/src/main/kotlin/net/jkcode/jkbenchmark/orm/mybatis/model/Department.java
package net.jkcode.jkbenchmark.orm.mybatis.model;

import java.util.List;

public class Department {
    private Integer id;
    private String title;
    private String intro;
    private List<Employee> emps;

    public Department(){

    }

    public Department(Integer id, String title, String intro) {
        this.id = id;
        this.title = title;
        this.intro = intro;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getIntro() {
        return intro;
    }

    public void setIntro(String intro) {
        this.intro = intro;
    }

    public List<Employee> getEmps() {
        return emps;
    }

    public void setEmps(List<Employee> emps) {
        this.emps = emps;
    }
}

File: orm/jkorm-mybatis/src/main/kotlin/net/jkcode/jkbenchmark/orm/jkorm/model/Department.kt
package net.jkcode.jkbenchmark.orm.jkorm.model

import net.jkcode.jkmvc.orm.* 
import java.util.*

/**
 * 部门
 *
 * @author shijianhang<772910474@qq.com>
 * @date 2021-04-13 14:18:07
 */
class Department(vararg pks: Any): Orm(*pks) {

	public constructor() : this(*arrayOf())

	// 伴随对象就是元数据
 	companion object m: OrmMeta(Department::class, "部门", "department", DbKeyNames("id")){
		init {
			hasMany("emps", Employee::class, "dep_id")
		}
	}

	// 代理属性读写
	public var id:Integer by property() //

	public var title:String by property() //  

	public var intro:String by property() //

	public var emps: List<Employee> by listProperty() // 有多个员工

}